fieldpath/element.go

```diff
@@ -24,6 +24,58 @@ import (
 	"sigs.k8s.io/structured-merge-diff/value"
 )
 
+type KeyValue struct {
+	Key   string
+	Value value.Value
+}
+
+func SortKeyValues(kv []KeyValue) {
+	if len(kv) < 2 {
+		return
+	}
+	if len(kv) == 2 {
+		if kv[1].Key < kv[0].Key {
+			kv[0], kv[1] = kv[1], kv[0]
+		}
+		return
+	}
+	sort.SliceStable(kv, func(i, j int) bool {
+		return kv[i].Key < kv[j].Key
+	})
+}
+
+func KeyValueLess(lhs, rhs []KeyValue) bool {
```

Comment:
This is actually comparing lists of keyvalue pairs, "KeyValueListLess"?

Reply:
> I'm curious if I should make a type for `[]KeyValue` and make that a method of the type.

And then I would make this a Less method of the `[]KeyValue` type.

Reply:
Maybe even do a `type KeyValueList []KeyValue`? And consider making this call work like `lhs.Less(rhs)`?